File: for_dev/gradus_anti_cheat.py
import random
from typing import Any, Dict, Union

class AntiCheat:
    def __init__(self, variables: Dict[str, Any] = None):
        print("[GRADUS-AC] Анти-чит создан, документация на официальном сайте")
        self.monitoring = False
        self.variables: Dict[str, dict] = variables if variables else {}
        self._randint = random.randint(10000, 99999)
        self._randfloat = random.randint(100000, 999999) / 10.0
        _words = ["cocoon", "melon", "apple", "orange", "banana", "pineapple", "grape", "juice"]
        self._randstr = random.choice(_words)

    def _sys_int(self, name: str, value: int) -> bool:
        self.variables[name] = {"value": value + self._randint, "type": "int"}
        return True

    def _sys_str(self, name: str, value: str) -> bool:
        self.variables[name] = {"value": value + self._randstr, "type": "str"}
        return True

    def _sys_bool(self, name: str, value: bool) -> bool:
        self.variables[name] = {"value": not value, "type": "bool"}
        return True

    def _sys_float(self, name: str, value: float) -> bool:
        self.variables[name] = {"value": value + self._randfloat, "type": "float"}
        return True

    def _sys_dict(self, name: str, value: dict) -> bool:
        self.variables[name] = {"value": value, "type": "dict"}
        return True

    def _sys_list(self, name: str, value: list) -> bool:
        self.variables[name] = {"value": value, "type": "list"}
        return True

    def _sys_other(self, name: str, value) -> bool:
        print(f"[GRADUS-AC] Неподдерживаемый тип у '{name}', поддерживаются: int, str, bool, float, dict, list")
        return False

    def start_monitoring(self) -> bool:
        self.monitoring = True
        print("[GRADUS-AC] Античит включён!")
        return True

    def stop_monitoring(self) -> bool:
        self.monitoring = False
        print("[GRADUS-AC] Античит выключен!")
        return True

    def add_variable(self, name: str = "VARIABLE", value=None) -> str:
        if name == "VARIABLE":
            name = f"VARIABLE{random.randint(10000, 99999)}"

        t = type(value)
        if t == int:
            self._sys_int(name, value)
        elif t == str:
            self._sys_str(name, value)
        elif t == bool:
            self._sys_bool(name, value)
        elif t == float:
            self._sys_float(name, value)
        elif t == dict:
            self._sys_dict(name, value)
        elif t == list:
            self._sys_list(name, value)
        else:
            self._sys_other(name, value)
            raise ValueError(f"Неподдерживаемый тип переменной '{name}'")
        return name

    def get_variable(self, name: str) -> Any:
        if name not in self.variables:
            return None
        var = self.variables[name]
        t = var["type"]
        v = var["value"]
        if t == "int":
            return v - self._randint
        elif t == "str":
            return v.replace(self._randstr, "")
        elif t == "bool":
            return not v
        elif t == "float":
            return v - self._randfloat
        elif t in ("dict", "list"):
            return v
        else:
            return None

    def is_hacked(self, name: str, value) -> bool:
        if not self.monitoring:
            return False
        stored = self.get_variable(name)
        if stored is None:
            return True
        return stored != value

File: for_dev/GradusAntiCheat.java
import java.util.*;

public class GradusAntiCheat {
    private boolean monitoring = false;
    private final Map<String, StoredVar> variables = new HashMap<>();
    private final int randInt;
    private final double randFloat;
    private final String randStr;

    private static class StoredVar {
        enum Type { INT, STR, BOOL, FLOAT, DICT, LIST }
        Type type;
        Object value;
        StoredVar(Type type, Object value) {
            this.type = type;
            this.value = value;
        }
    }

    public GradusAntiCheat() {
        System.out.println("[GRADUS-AC] Анти-чит создан, документация на официальном сайте");
        Random rng = new Random();
        this.randInt = 10000 + rng.nextInt(90000);
        this.randFloat = (100000 + rng.nextInt(900000)) / 10.0;
        String[] words = {"cocoon", "melon", "apple", "orange", "banana", "pineapple", "grape", "juice"};
        this.randStr = words[rng.nextInt(words.length)];
    }

    private boolean sysInt(String name, int value) {
        variables.put(name, new StoredVar(StoredVar.Type.INT, value + randInt));
        return true;
    }
    private boolean sysStr(String name, String value) {
        variables.put(name, new StoredVar(StoredVar.Type.STR, value + randStr));
        return true;
    }
    private boolean sysBool(String name, boolean value) {
        variables.put(name, new StoredVar(StoredVar.Type.BOOL, !value));
        return true;
    }
    private boolean sysFloat(String name, double value) {
        variables.put(name, new StoredVar(StoredVar.Type.FLOAT, value + randFloat));
        return true;
    }
    private boolean sysDict(String name, Map<String, Object> value) {
        variables.put(name, new StoredVar(StoredVar.Type.DICT, value));
        return true;
    }
    private boolean sysList(String name, List<Object> value) {
        variables.put(name, new StoredVar(StoredVar.Type.LIST, value));
        return true;
    }
    private boolean sysOther(String name) {
        System.out.println("[GRADUS-AC] Неподдерживаемый тип у \"" + name + "\", поддерживаются: int, str, bool, float, dict, list");
        return false;
    }

    public boolean startMonitoring() {
        monitoring = true;
        System.out.println("[GRADUS-AC] Античит включён!");
        return true;
    }

    public boolean stopMonitoring() {
        monitoring = false;
        System.out.println("[GRADUS-AC] Античит выключен!");
        return true;
    }

    public String addVariable(String name, Object value) {
        if (name == null || name.equals("VARIABLE")) {
            name = "VARIABLE" + (10000 + new Random().nextInt(90000));
        }

        if (value instanceof Integer) {
            sysInt(name, (Integer) value);
        } else if (value instanceof String) {
            sysStr(name, (String) value);
        } else if (value instanceof Boolean) {
            sysBool(name, (Boolean) value);
        } else if (value instanceof Double || value instanceof Float) {
            double v = (value instanceof Double) ? (Double) value : ((Float) value).doubleValue();
            sysFloat(name, v);
        } else if (value instanceof Map) {
            @SuppressWarnings("unchecked")
            Map<String, Object> map = (Map<String, Object>) value;
            sysDict(name, map);
        } else if (value instanceof List) {
            @SuppressWarnings("unchecked")
            List<Object> list = (List<Object>) value;
            sysList(name, list);
        } else {
            sysOther(name);
            throw new IllegalArgumentException("Неподдерживаемый тип переменной '" + name + "'");
        }
        return name;
    }

    public Object getVariable(String name) {
        StoredVar var = variables.get(name);
        if (var == null) return null;

        switch (var.type) {
            case INT:
                return ((Integer) var.value) - randInt;
            case STR:
                return ((String) var.value).replace(randStr, "");
            case BOOL:
                return !((Boolean) var.value);
            case FLOAT:
                return ((Double) var.value) - randFloat;
            case DICT:
            case LIST:
                return var.value;
            default:
                return null;
        }
    }

    public boolean isHacked(String name, Object value) {
        if (!monitoring) return false;
        Object stored = getVariable(name);
        if (stored == null) return true;
        return !stored.equals(value);
    }
}